### ISSUE TITLE
The application unexpectedly exited. Diagnostic information has been dumped to the JavaScript error console.

### ISSUE THREAD

**Issue by Make42** [rstudio/shiny#864]
My shiny application runs fine in RStudio.
Shiny Server also runs fine, as I can see in http://localhost:3838
I copied the app folder with its content to /srv/shiny-server.
I can see in my browser that part of the app is loaded, but part is not and I get the message:

      The application unexpectedly exited.
      Diagnostic information has been dumped to the JavaScript error console.

I commented out all the leaflet-stuff and did

      $ sudo restart shiny-server

I still get the message.

What's up with that?

**Comment by jcheng5** [comment]
Nothing else printed in the JS console at all? Do you see anything in /var/log/shiny-server/?

**Comment by Make42** [comment]
I do not know how to access the "JS console". I don't get an error message when running the app in RStudio. So how do I check the "JS console"?

I wasn't aware that logs are saved in /var/log/shiny-server - I will check it out first thing in the morning tomorrow. (Here it is evening already.)

**Comment by yihui** [comment]
@Make42 The console is in your web browser, e.g. Chrome: https://developer.chrome.com/devtools/docs/console Firefox: https://developer.mozilla.org/en-US/docs/Tools/Web_Console

**Comment by Make42** [comment]
Thanks for the links @yihui - I was able to debug my code.

However, the issue https://github.com/rstudio/shiny/issues/863 returned for the shiny-server.

      sudo su -   -c "R -e \"devtools::install_github('rstudio/shiny')\""

does not help here. I guess shiny-server itself does not use the R shiny package, but some code which still contains the bug. I installed shiny server via

      http://rstudio.github.io/shiny-server/latest/#ubuntu-12.04

Should I install via https://github.com/rstudio/shiny-server ? It explicitly states "experimental", and I got to run it in production.
